src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs

```diff
@@ -369,73 +276,140 @@ private static bool IsDatabaseBeingDeleted(string tag, RawDatabaseRecord databas
+            foreach (var route in routes)
             {
                 token.ThrowIfCancellationRequested();
+                
+                await InvokeAndWriteToArchive(archive, jsonOperationContext, localEndpointClient, route, path, endpointParameters, token);
+            }
+        }
 
-                try
-                {
-                    var entry = archive.CreateEntry(DebugInfoPackageUtils.GetOutputPathFromRouteInformation(route, path ?? databaseName));
-                    entry.ExternalAttributes = ((int)(FilePermissions.S_IRUSR | FilePermissions.S_IWUSR)) << 16;
+        internal static async Task InvokeAndWriteToArchive(ZipArchive archive, JsonOperationContext jsonOperationContext, LocalEndpointClient localEndpointClient,
+            RouteInformation route, string path,
+            Dictionary<string, Microsoft.Extensions.Primitives.StringValues> endpointParameters = null,
+            CancellationToken token = default)
+        {
+            try
+            {
+                var response = await localEndpointClient.InvokeAsync(route, endpointParameters);
 
-                    using (var entryStream = entry.Open())
-                    using (var writer = new BlittableJsonTextWriter(jsonOperationContext, entryStream))
+                var entryName = DebugInfoPackageUtils.GetOutputPathFromRouteInformation(route, path, response.ContentType == "text/plain" ? "txt" : "json");
+                var entry = archive.CreateEntry(entryName);
+                entry.ExternalAttributes = ((int)(FilePermissions.S_IRUSR | FilePermissions.S_IWUSR)) << 16;
+
+                using (var entryStream = entry.Open())
+                {
+                    if (response.ContentType == "text/plain")
                     {
-                        using (var endpointOutput = await localEndpointClient.InvokeAndReadObjectAsync(route, jsonOperationContext, endpointParameters))
+                        await response.Body.CopyToAsync(entryStream, token);
+                    }
+                    else
+                    {
+                        using (var writer = new BlittableJsonTextWriter(jsonOperationContext, entryStream))
                         {
+                            var endpointOutput = jsonOperationContext.ReadForMemory(response.Body, $"read/local endpoint/{route.Path}");
                             jsonOperationContext.Write(writer, endpointOutput);
                             writer.Flush();
-                            await entryStream.FlushAsync(token);
                         }
                     }
+                    await entryStream.FlushAsync(token);
                 }
-                catch (OperationCanceledException)
-                {
-                    throw;
-                }
-                catch (Exception e)
+            }
+            catch (OperationCanceledException)
+            {
+                throw;
+            }
+            catch (InvalidStartOfObjectException e)
```

Comment:
Not a good idea. We _always_ want to get the results, even if we had an error.
As it currently stands, this sort of failure would prevent us from generating the debug package, which is a Bad Idea.

Keep the change on the `txt`, but we _must_ be able to proceed, regardless of errors.

Reply:
Didn't catch that :) pushed a fix now